base/operators.jl

```diff
@@ -48,26 +71,23 @@ Scalar types generally do not need to implement `isequal` separate from `==`, un
 represent floating-point numbers amenable to a more efficient implementation than that
 provided as a generic fallback (based on `isnan`, `signbit`, and `==`).
 """
-isequal(x, y) = x == y
+function isequal(x, y)
+    @inbounds res = x == y
+    res
+end
+
+# FIXME: Required since @inbounds does seem to work during early bootstrap
+isequal(::Void, ::Void) = true
+isequal(::Void, ::Any) = false
+isequal(::Any, ::Void) = false
+isequal(x::Union{Method, TypeName}, y::Union{Method, TypeName}) = x === y
```

Comment:
Unfortunately, it's hard to track since the failure happens during bootstrap, and the backtrace doesn't mention the function name (even with `julia-debug`). Is there anything I can do to improve that?